src/IO/HashingReadBuffer.h

```diff
@@ -18,29 +18,38 @@ class HashingReadBuffer : public IHashingBuffer<ReadBuffer>
     {
         working_buffer = in.buffer();
         pos = in.position();
+        hashing_begin = pos;
+    }
 
-        /// calculate hash from the data already read
-        if (!working_buffer.empty())
+    uint128 getHash()
```

Comment:
I need hash calculation to be up to the current position, not up to the end of the buffer.
Otherwise `hashing_read_buffer.ignore(prefix_size)` (see BackupImpl.cpp) doesn't work properly.